servers/brave_search/tests/test_brave_search.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+import httpx
+from unittest.mock import MagicMock, Mock, patch, AsyncMock
+from brave_search_mcp.main import (
+    brave_web_search, 
+    fetch_website, 
+    make_brave_request, 
+)
+
+# Sample response data for mocking
+WEB_SEARCH_RESPONSE = {
+    "web": {
+        "results": [
+            {
+                "title": "Test Result 1",
+                "url": "https://example.com/1",
+                "description": "This is test result 1"
+            },
+            {
+                "title": "Test Result 2",
+                "url": "https://example.com/2",
+                "description": "This is test result 2"
+            }
+        ]
+    }
+}
+
+ERROR_RESPONSE = {"error": "Test error message"}
+
+# HTML content for website fetch testing
+HTML_CONTENT = """
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Test Page</title>
+    <style>
+        body { font-family: Arial; }
+    </style>
+</head>
+<body>
+    <h1>Test Page Heading</h1>
+    <p>This is a paragraph.</p>
+    <script>
+        console.log("This should be removed");
+    </script>
+</body>
+</html>
+"""
+
+EXPECTED_TEXT = "Test Page Heading\nThis is a paragraph."
+
+@pytest.mark.asyncio
+async def test_make_brave_request_success():
+    # Mock a successful API response
+    mock_response = AsyncMock(spec=httpx.Response)
+    mock_response.status_code = 200
+    mock_response.headers = {"Content-Type": "application/json"}
+    mock_response.json = MagicMock(return_value=WEB_SEARCH_RESPONSE)
+    
+    with patch("httpx.AsyncClient.get", return_value=mock_response):
+        result = await make_brave_request("https://api.example.com/test")
+        
+    assert result == WEB_SEARCH_RESPONSE
+    
+@pytest.mark.asyncio
+async def test_make_brave_request_http_error():
+    # Mock an HTTP error
+    mock_response = AsyncMock()
+    mock_error = httpx.HTTPStatusError(
+        "Error", 
+        request=AsyncMock(), 
+        response=AsyncMock()
+    )
+    mock_error.response.status_code = 401
+    mock_response.raise_for_status = MagicMock(side_effect=mock_error)
+    
+    with patch("httpx.AsyncClient.get", return_value=mock_response):
+        result = await make_brave_request("https://api.example.com/test")
+        
+
+    print("this is the result", result)
```

Comment:
```suggestion
    print("this is the result", result)
```
I think we can omit the print statements in the tests as they might pollute the test execution logs :-)